pandas/core/generic.py

```diff
@@ -6065,19 +6071,43 @@ def asof(self, where, subset=None):
 
         Returns
         -------
-        where is scalar
+        `where` is scalar
 
-          - value or NaN if input is Series
-          - Series if input is DataFrame
+            - value or `NaN` if input is Series
+            - Series if input is DataFrame
 
-        where is Index: same shape object as input
+        `where` is :class:`pandas.Index`:
+            - same shape object as input
 
         See Also
         --------
-        merge_asof
+        merge_asof : Perform an asof merge. Similar to left join.
 
+        Examples
+        --------
+        Take all columns into consideration
```

Comment:
Can you add additional examples for

- Series w/ scalar where
- Series w/ array-like where
- DataFrame w/ scalar where